src/umzug.ts

```diff
@@ -375,7 +388,7 @@ export class Umzug extends EventEmitter {
 	*/
 	private async _findMigrations(): Promise<Migration[]> {
 		if (Array.isArray(this.options.migrations)) {
-			return this.options.migrations;
+			return this.options.migrations.sort((a, b) => this.options.migrationSorting(a.file, b.file));
```

Comment:
Since `sort` operates in-place, this changes the value of `this.options.migrations` - is this intended? If not, please use `.slice().sort()` instead.

Reply:
In fact, I do want to change the order of the value of `this.options.migrations`. In my case, I'm using the version as the filename, such as `1.0.0`. So I want to make it run the migration script with the correct order.

Is there any issue or side effect if we change the value of `this.options.migrations`?

Reply:
I don't think altering it directly is a good idea, I wouldn't expect a class to modify a value I passed into it. In the future I hope to enable the [prefer-readonly-parameter-types rule](https://github.com/sequelize/umzug/blob/69d7b571e66680b93218599a868f824969daefb6/.eslintrc.js#L55).

Please try to apply your change in [this line](https://github.com/sequelize/umzug/blob/fe2bb23b2499d750fa6461a7487e15644fffe359/src/umzug.ts#L32) instead, using `.slice().sort()`. Thanks!

Reply:
Fixed. Thanks.